### ISSUE TITLE
Bindings call setters of bound objects unnecessarily

### ISSUE THREAD

**Issue by papafe** [dotnet/maui#12001]
### Description

Binding `ItemSource` of a `CollectionView` or `ListView` to a collection of some objects results in the setters of the objects' properties being called unnecessarily when the views appear on screen. The values on the properties are not changed, as the setters are called with the same value, but it seems an unnecessary action. 

I was able to reproduce this in both iOS and Android, didn't try other platforms. 

### Steps to Reproduce

Run the reproduction project repository. 

The `CollectionView` is bound to a list of `Item`
``` csharp
    public class Item
    {
        private string name;

        private bool isComplete;

        public string Name
        {
            get => name;
            set
            {
                OnPropChange(name, value);
            }
        }

        public bool IsComplete
        {
            get => isComplete;
            set
            {
                OnPropChange(IsComplete, value);
            }
        }

        private void OnPropChange(object oldVal, object newVal, [CallerMemberName] string propName = "")
        {
            Console.WriteLine($"PropChanged: {propName}: {oldVal} -> {newVal}");
        }
    }
```

When the main page of the app is opened the setter of both properties is called several times unnecessarily, and you will be able to see `OnPropChange` printing to console the old and new value of the property as well as the property name. 

This happens also if a `ListView` is used, even though the number of times the setter gets called is different. 

### Link to public reproduction project repository

https://github.com/papafe/test-maui

Much simpler repro: https://github.com/mattleibow/maui-bug-repros/tree/repro-lots-of-updates

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

### Affected platform versions

iOS 16, Android api 33

**Comment by papafe** [comment]
@jfversluis Yes, if you change the `Entry` to a `Label`, then the setter will not be called. 
Sorry, I think I was not clear in my message. The issue here is that the setters of the properties are being called without user interaction. It seems that MAUI UI framework is calling the setters when the bindings are being setup, or at least that's what it looks like if I check the call stack. This looks like erroneous behaviour, as I do not think it is expected that the setters are called without user and/or developer code interaction.

**Comment by mattleibow** [comment]
Did some testing and this appears to be the binding engine. Not sure if this is good or bad.

I created a much simple repro here: https://github.com/mattleibow/maui-bug-repros/tree/repro-lots-of-updates

```xaml
<Entry Text="{Binding Name}" />
```

```cs
public partial class MainPage : ContentPage
{
    private string name = "This is text.";

    public MainPage()
    {
        InitializeComponent();

        BindingContext = this;
    }

    public string Name
    {
        get => name;
        set
        {
            System.Diagnostics.Debug.WriteLine($"Name fired: '{name}' => '{value}'");

            name = value;
        }
    }
}
``

Running this app results in this in the output:

```
Name fired: 'This is text.' => 'This is text.'
```

**Comment by jfversluis** [comment]
@StephaneDelcroix

**Comment by Ying-6** [comment]
This issue has been verified using Visual Studio 17.14 Preview 1.1(9.0.40) release mode. Not repro this issue on android platform.
![Image](https://github.com/user-attachments/assets/ef6369fa-4880-4740-8da6-3ddb0fcbf1d7)
https://github.com/mattleibow/maui-bug-repros.git
